modules/router/templates/assets_origin.conf.erb

```diff
@@ -49,7 +49,16 @@ server {
 
   <%- @asset_manager_uploaded_assets_routes.each do |path| -%>
   location <%= path %> {
-    proxy_pass <%= @upstream_ssl ? 'https' : 'http' %>://static.<%= @app_domain %>;
+    # Explicitly re-include Strict-Transport-Security header, this
+    # forces nginx not to clear Cache-Control headers further up the
+    # stack.
+    include /etc/nginx/add-sts.conf;
+
+    add_header "Access-Control-Allow-Origin" "*";
+    add_header "Access-Control-Allow-Methods" "GET, OPTIONS";
+    add_header "Access-Control-Allow-Headers" "origin, authorization";
```

Comment:
Minor: Did you consider extracting this into an `include` to avoid the duplication?

Reply:
I've had a quick look at introducing an `include` to reduce the duplication but think it might be worth leaving this for another time. There are a few different `Access-Control-Allow-*` headers/values being set and I think it might be worth seeing whether we can make them consistent before removing the duplication.